### PR TITLE
Check results bash script tests

### LINKED ISSUE #6: Implement CLI tests in TravisCI for check-results
See https://github.com/nsls-ii-forge/nsls2forge-utils/pull/5#pullrequestreview-439200285.

### Test 1:
- create a conda environment from the `defaults` channel
- check there are no packages from e.g. `conda-forge` and verify the exit code is 0

### Test 2
- create a conda environment from the `conda-forge` channel
- check the channels with the `conda-forge` "forbidden" channel and verify the exit code is 1

### Test 3
- create a conda environment from the `conda-forge` channel
- check the channels with the `conda-forge` "forbidden" channel and the flag to ignore the exception and verify the exit code is 0

### PR DESCRIPTION
**Pull request by thomashopkins32**
Added tests for `check-results -t channels` command with various flags.

Closes #6

### CODE REVIEW

File: nsls2forge_utils/tests/test_check_results.sh [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+source ~/mc/etc/profile.d/conda.sh
+
+# test for default behavior (no packages installed)
+conda create --name df -y
+conda activate df
+conda install -c defaults pip -y
+pip install .
+check-results -t channels
+EXIT="$?"
+if [ $EXIT -ne "0" ]; then
```
Comment (mrakitin) [inline nsls2forge_utils/tests/test_check_results.sh:16]:
```suggestion
if [ "$EXIT" -gt "0" ]; then
```

File: nsls2forge_utils/tests/test_check_results.sh [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  echo "ERROR: default behavior test failed with exit code $EXIT"
+  exit 1
+fi
+
+# test for default channel behavior
+conda install -c defaults python
+check-results -t channels -f conda-forge
+EXIT="$?"
+if [ $EXIT -ne "0" ]; then
```
Comment (mrakitin) [inline nsls2forge_utils/tests/test_check_results.sh:25]:
```suggestion
if [ "$EXIT" -gt "0" ]; then
```

File: nsls2forge_utils/tests/test_check_results.sh [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+# tests for conda-forge channel + ignore
+conda create --name cf -y
+conda activate cf
+FORBIDDEN_NAME="conda-forge"
+conda install -c $FORBIDDEN_NAME --override-channels pip -y
+pip install .
+check-results -t channels -f $FORBIDDEN_NAME
+EXIT="$?"
+if [ $EXIT -eq "0" ]; then
```
Comment (mrakitin) [inline nsls2forge_utils/tests/test_check_results.sh:39]:
```suggestion
if [ "$EXIT" -eq "0" ]; then
```

File: nsls2forge_utils/tests/test_check_results.sh [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+  echo "ERROR: forbidden channel $FORBIDDEN_NAME raise error test failed with exit code $EXIT"
+  exit 1
+fi
+check-results -t channels -f $FORBIDDEN_NAME -i
```
Comment (mrakitin) [inline nsls2forge_utils/tests/test_check_results.sh:43]:
```suggestion

check-results -t channels -f $FORBIDDEN_NAME -i
```

File: nsls2forge_utils/tests/test_check_results.sh [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+EXIT="$?"
+if [ $EXIT -ne "0" ]; then
```
Comment (mrakitin) [inline nsls2forge_utils/tests/test_check_results.sh:45]:
```suggestion
if [ "$EXIT" -gt "0" ]; then
```

File: .travis.yml [hunk 6]
```diff
@@ -18,3 +18,5 @@ script:
   - codecov  # Upload the report to codecov.
   - flake8  # Enforce code style ('relaxed' line length limit is set in .flake8 config file).
   - make -C docs html  # Build the documentation.
+  - chmod +x ./nsls2forge_utils/tests/test_check_results.sh  # Add permissions
+  - ./nsls2forge_utils/tests/test_check_results.sh  # Run testing script
```
Comment (mrakitin) [inline .travis.yml:22]: Could this block of shell tests be moved right after the pytest execution (line 16)? That way we won't build the documentation if the test fails.

File: nsls2forge_utils/tests/test_check_results.sh [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+pip install .
+check-results -t channels
+EXIT="$?"
+if [ "$EXIT" -gt "0" ]; then
+  echo "ERROR: default behavior test failed with exit code $EXIT"
+  exit 1
+fi
+
+# test for default channel behavior
+conda install -c defaults python
```
Comment (mrakitin) [inline nsls2forge_utils/tests/test_check_results.sh:24]:
```suggestion
conda install -c defaults numpy  # this should bring a new dependency
```

File: .travis.yml [hunk 8]
```diff
@@ -6,11 +6,33 @@ cache:
...
+        arch="MacOSX"
+    else
+        echo "Unknown arch $TRAVIS_OS_NAME"
+        exit 1
+    fi
+    wget https://repo.continuum.io/miniconda/Miniconda3-latest-${arch}-x86_64.sh -O miniconda.sh
+    chmod +x miniconda.sh
+    ./miniconda.sh -b -p ~/mc
+    source ~/mc/etc/profile.d/conda.sh
+    conda update conda --yes
```
Comment (mrakitin) [inline .travis.yml:24]: This step may take some time. I think it's OK to not do it as we are going to use the latest distribution available.

File: .travis.yml [hunk 9]
```diff
@@ -6,11 +6,33 @@ cache:
...
+        arch="Linux"
+    elif [ "$TRAVIS_OS_NAME" == "osx" ]; then
+        arch="MacOSX"
+    else
+        echo "Unknown arch $TRAVIS_OS_NAME"
+        exit 1
+    fi
+    wget https://repo.continuum.io/miniconda/Miniconda3-latest-${arch}-x86_64.sh -O miniconda.sh
+    chmod +x miniconda.sh
+    ./miniconda.sh -b -p ~/mc
```
Comment (mrakitin) [inline .travis.yml:22]:
```suggestion
    bash miniconda.sh -b -p ~/mc
```

File: .travis.yml [hunk 10]
```diff
@@ -6,11 +6,33 @@ cache:
...
+    if [ "$TRAVIS_OS_NAME" == "linux" ]; then
+        arch="Linux"
+    elif [ "$TRAVIS_OS_NAME" == "osx" ]; then
+        arch="MacOSX"
+    else
+        echo "Unknown arch $TRAVIS_OS_NAME"
+        exit 1
+    fi
+    wget https://repo.continuum.io/miniconda/Miniconda3-latest-${arch}-x86_64.sh -O miniconda.sh
+    chmod +x miniconda.sh
```
Comment (mrakitin) [inline .travis.yml:21]:
```suggestion
```

File: .travis.yml [hunk 11]
```diff
@@ -6,11 +6,33 @@ cache:
...
+    ./miniconda.sh -b -p ~/mc
+    source ~/mc/etc/profile.d/conda.sh
+    conda update conda --yes
+    export CONDARC=ci/condarc
```
Comment (mrakitin) [inline .travis.yml:25]: I don't think we need this line. I guess we don't have the `ci` dir here and probably do not want to.

File: .travis.yml (hunk 8 above)
Comment (thomashopkins32) [reply .travis.yml:24]: You are talking about line 24 here correct?
`conda update conda --yes`

Comment (mrakitin) [reply .travis.yml:24]: Yes, just this one line.

**Pull request merged**